### ISSUE TITLE
Allow user to define snapshot_every_n_steps for native dataloaders, aka StatefulDataloader

### ISSUE THREAD

**Issue by LiuzcEECS** [ByteDance-Seed/VeOmni#487]
### Feature request

In build_native_dataloader, we are building DistributedDataloader, which is a subclass of StatefulDataloader.
The StatefulDataloader has snapshot_every_n_steps as an arg, which controls how freq the StatefulDataloader requests snapshotting from upstream datasets. The default value is 1, and we don't define it explicitly inthe  VeOmni library, which caused the upstream dataset to call state_dict() for each step even without saving.
I suggest we allow the users to define the value with a larger number and then in the training scripts, when saving, instead of defining the saving state as
```
"extra_state": {
                        "global_step": global_step,
                        "lr_scheduler": lr_scheduler.state_dict(),
                        "train_dataloader": train_dataloader.state_dict(),
                        "environ_meter": environ_meter.state_dict(),
                    },
```
We need to find a way to record the accurate "global_step" since the train_dataloader's state might be fetched from snapshots a few steps ago and the local global_step variable is different than the real "global_step" when the train_dataloader is snapshoted

### Motivation

To reduce the overhead of calling state_dict() for each step

### Your contribution

I can contribute a PR to

- add an argument of `snapshot_every_n_steps`
- update all in-house training scripts to with a more accurate way of recording global_steps
